classes/sota_raspberrypi.bbclass

```diff
@@ -1,11 +1,20 @@
 RPI_USE_U_BOOT_sota = "1"
-KERNEL_IMAGETYPE_sota = "uImage"
+
+KERNEL_CLASSES_append_sota = " kernel-fitimage"
+KERNEL_IMAGETYPE_sota = "fitImage"
+
 PREFERRED_PROVIDER_virtual/bootloader_sota ?= "u-boot"
-UBOOT_MACHINE_raspberrypi2_sota ?= "rpi_2_defconfig"
-UBOOT_MACHINE_raspberrypi3_sota ?= "rpi_3_32b_defconfig"
+UBOOT_ENTRYPOINT_sota ?= "0x00008000"
 
 IMAGE_FSTYPES_remove_sota = "rpi-sdimg"
 OSTREE_BOOTLOADER ?= "u-boot"
 
 # OSTree puts its own boot.scr to bcm2835-bootfiles
```

Comment:
Please update this comment as well, since boot.scr is not removed anymore.

Reply:
@rsalveti Yes, probably, but I see some problems here
1. Aktualizr doesn't currently support updating the bootloader, and I don't know how soon it will.
2. Even if it supported updating the boot partition, the synchronization problem would arise. If I updated my kernel/dtb with a new version where support for some parameters was dropped, while others have been added, how do I make sure that the new kernel is always used with the new bootloader? It gets even more complicated once aktualizr supports rollbacks.

So my idea (might be a wrong one of course) was that user doesn't update bootloader to enable new functionality, it is all done with kernel/filesystem updates (which are synchronized by OSTree). Bootloader update when it is supported should only be used to fix bugs/vulnerabilities in bootloader itself.

Reply:
Yeah, looking again it seems that the only really safe way to handle this with OSTree is by using fit format for devices requiring device tree.

It will make it incompatible with the traditional config.txt rpi file (e.g. when adding a new device tree overlay), but it is indeed a lot better to let the system handle what is supported or not.